synapse/handlers/e2e_keys.py

```diff
@@ -1450,19 +1450,25 @@ async def _retrieve_cross_signing_keys_for_remote_user(
 
         return desired_key_data
 
-    async def is_cross_signing_set_up_for_user(self, user_id: str) -> bool:
+    async def check_cross_signing_setup(self, user_id: str) -> Tuple[bool, bool]:
         """Checks if the user has cross-signing set up
 
         Args:
             user_id: The user to check
 
-        Returns:
-            True if the user has cross-signing set up, False otherwise
+        Returns: a 2-tuple of booleans
+            - whether the user has cross-signing set up, and
+            - whether the user's master cross-signing key may be replaced without UIA.
```

Comment:
A bit naughty, since there aren't really 4 states here. But I think this makes things clearer at the call site.